### PR TITLE
From sdk4 sup

### PR DESCRIPTION
**Pull request by matemax**

### CODE REVIEW

File: lunavl/sdk/estimators/face_estimators/credibility_check.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+from lunavl.sdk.errors.errors import LunaVLError
+from lunavl.sdk.errors.exceptions import CoreExceptionWrap, LunaSDKException
+from ..base import BaseEstimator
+from ..face_estimators.facewarper import FaceWarp, FaceWarpedImage
+from ...base import BaseEstimation
+
+
+class CredibilityCheck(BaseEstimation):
+    """
+    Structure credibility check
```
Comment (matemax) [inline lunavl/sdk/estimators/face_estimators/credibility_check.py]: credibility score?

File: tests/test_estimate_descriptor.py [hunk 2]
```diff
@@ -400,7 +400,7 @@ def test_extract_descriptors_batch_positive_and_negative(self):
                                 exceptionInfo.value.context[1], LunaVLError.InvalidImageSize
                             )
 
-    @pytest.mark.skip(msg="Skip error test")  # TODO: Сheck why the test failed
+    @pytest.mark.skip(msg="Skip error test")  # TODO: SDK return isError=False. Not raises LunaSDKException
```
Comment (matemax) [inline tests/test_estimate_descriptor.py:403]: ?

File: lunavl/sdk/detectors/facedetector.py [hunk 3]
```diff
@@ -365,5 +365,7 @@ def redetect(
         return res
 
     def setDetectionComparer(self):
-        """"""
+        """
+        set detection comparer
```
Comment (ddc67cd) [inline lunavl/sdk/detectors/facedetector.py]: Pretty useless.

**Comment by ddc67cd** [comment]
Docstrings should follow pep-257 (https://www.python.org/dev/peps/pep-0257/).

> The docstring is a phrase ending in a **period**. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ...".

File: lunavl/sdk/estimators/face_estimators/credibility.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+    Estimation properties:
+
+        - credibility
+    """
+
+    #  pylint: disable=W0235
+    def __init__(self, credibility: CredibilityCheckEstimation):
+        """
+        Init.
```
Comment (ddc67cd) [inline lunavl/sdk/estimators/face_estimators/credibility.py]: Useless doc, useless method.

File: lunavl/sdk/luna_faces.py [hunk 5]
```diff
@@ -28,11 +30,37 @@
...
+    Settings for detection
+
+    Attributes:
+        estimateMaskFromDetection: estimate mask from detection or warp
+
+    """
+
+    __slots__ = "_estimateMaskFromDetection"
+
+    def __init__(self, estimateMaskFromDetection: bool = False):
```
Comment (ddc67cd) [inline lunavl/sdk/luna_faces.py]: This looks like a boilerplate. Using frozen dataclass + slots requires less code.

File: lunavl/sdk/luna_faces.py [hunk 6]
```diff
@@ -198,6 +250,17 @@ def glasses(self) -> Glasses:
             self._glasses = self.estimatorCollection.glassesEstimator.estimate(self.warp)
         return self._glasses
 
+    @property
+    def credibility(self) -> Credibility:
```
Comment (ddc67cd) [inline lunavl/sdk/luna_faces.py:240]: Would be good to eventually replace this boilerplate with decorator cached_property (its builtin in 3.8 and has backport to older versions)

**Review by ddc67cd — CHANGES_REQUESTED** [review]

File: lunavl/sdk/luna_faces.py [hunk 7]
```diff
@@ -1,4 +1,5 @@
-"""Module realize hight level api for estimate face attributes
+"""
+Module realize hight level api for estimate face attributes
```
Comment (ddc67cd) [inline lunavl/sdk/luna_faces.py]: `High-level api for estimating face attributes`

File: lunavl/sdk/luna_faces.py (hunk 5 above)
Comment (matemax) [reply lunavl/sdk/luna_faces.py]: good idea, but this project support python3.6 now. I don't think that need to break support 3.6 for this feature.

File: lunavl/sdk/luna_faces.py (hunk 6 above)
Comment (matemax) [reply lunavl/sdk/luna_faces.py:240]: I will create other mr with `cached_property`

File: lunavl/sdk/luna_faces.py (hunk 5 above)
Comment (matemax) [reply lunavl/sdk/luna_faces.py]: ++

File: lunavl/sdk/estimators/face_estimators/credibility.py (hunk 4 above)
Comment (matemax) [reply lunavl/sdk/estimators/face_estimators/credibility.py]: remove

File: lunavl/sdk/detectors/facedetector.py (hunk 3 above)
Comment (matemax) [reply lunavl/sdk/detectors/facedetector.py]: remove

File: lunavl/sdk/luna_faces.py (hunk 7 above)
Comment (matemax) [reply lunavl/sdk/luna_faces.py]: ++

**Pull request merged**